### PR TITLE
Implement locale selector in React page explorer

### PR DESCRIPTION
**Pull request by kaedroho**
This PR implements a locale switcher in the react page explorer.

image

Based on: https://github.com/wagtail/wagtail/pull/6488

### CODE REVIEW

**Comment by kaedroho** [comment]
To simplify the implementation of this PR, I've removed the filter when browsing the root page. It now instead displays the language next to each page:
image

The selector still appears when you navigate into a page

**Comment by kaedroho** [comment]
Rebased on top of https://github.com/wagtail/wagtail/pull/6488

File: client/src/components/Explorer/ExplorerItem.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ const ExplorerItem: React.FunctionComponent<ExplorerItemProps> = ({ item, onClic
   const { id, admin_display_title: title, meta } = item;
   const hasChildren = meta.children.count > 0;
   const isPublished = meta.status.live && !meta.status.has_unpublished_changes;
+  const localeName = meta.parent?.id === 1 && meta.locale && (LOCALE_NAMES.get(meta.locale) || meta.locale);
```
Comment (allcaps) [inline client/src/components/Explorer/ExplorerItem.tsx:29]: I don't know this syntax, `?.`. But I guess it is to check if `parent` exists and if so, get the `id`?

Comment (kaedroho) [reply client/src/components/Explorer/ExplorerItem.tsx:29]: Yep, that's right. ``meta.parent?.id`` is equivilant to ``meta.parent && meta.parent.id``

**Review by allcaps — APPROVED** [review]
LGTM

**Comment by kaedroho** [comment]
Thanks @allcaps!

Rebased on master

**Pull request merged**